packages/preferences/src/browser/preferences-editor.tsx

```diff
@@ -0,0 +1,567 @@
+                preferenceId={id}
+                value={JSON.stringify(value, undefined, 2)}
+                onSubmit={this.setPreference}
+                data-preference-id={id}
+            />
+        );
+    }
+
+    renderKeyValueInputs(id: string, item: any): JSX.Element {
+        let value: string;
+        if (isEmpty(item)) {
+            value = '{}';
+        } else {
+            const objectToRender = Object.keys(item).reduce((prev: any, key: string) => {
+                const total = prev;
+                if (typeof item[key] === 'object') {
+                    total[key] = item[key].default !== undefined ? item[key].default : 'undefined';
+                } else {
+                    total[key] = item[key];
+                }
+                return total;
+            }, {});
+
+            value = JSON.stringify(objectToRender, undefined, 2);
+        }
+
+        return (
+            <PreferencesObjectInput
+                preferenceId={id}
+                value={value}
+                onSubmit={this.submitObjectInput}
+                data-preference-id={id}
+            />
+        );
+    }
+
+    submitObjectInput = (id: string, value: object): void => {
+        const cleanObjectValue = pickBy(value, val => ['undefined', 'default'].indexOf(val) === -1);
+        this.setPreference(id, cleanObjectValue);
+    };
+
+    renderArrayInput(preference: PreferenceDataProperty): React.ReactNode {
+        const { id } = preference;
+        const values: string[] = this.getPreference(id).value;
+        return (
+            <PreferencesArray
+                preferenceID={id}
+                values={values}
+                removeValue={this.removeArrayPreferenceValueByIndex}
+                submit={this.submitArrayInput}
+            />
+        );
+    }
+
+    submitArrayInput = (id: string, value: string): void => {
+        const { value: before, items: { type } } = this.getPreference(id);
+        const typedValue = this.castStringToType(type, value);
+        if (value && Array.isArray(before) && typedValue !== undefined) {
+            this.setPreference(id, [...this.getPreference(id).value, typedValue]);
+        } else {
+            this.messageService.warn(`${value} is not a valid ${type}`);
+        }
+    };
+
+    castStringToType(type: string, value: string): string | number | boolean | undefined {
+        try {
+            const num = Number(value);
+
+            switch (type) {
+                case 'number':
+                    return !Number.isNaN(num) ? num : undefined;
+                case 'boolean':
+                    if (value === 'true') {
+                        return true;
+                    } if (value === 'false') {
+                        return false;
+                    }
+                    break;
+                default:
+                    return value;
+            }
+        } catch (e) {
+            console.error(e); // eslint-disable-line no-console
```

Comment:
console should be allowed, not sure why `eslint-disable-line no-console` is required?

Reply:
I will remove it. Thanks for the heads up.